fplus-lib/src/core/mod.rs

```diff
@@ -3524,6 +3533,46 @@ _The initial issue can be edited in order to solve the request of the verifier.
 
         Ok(updated_application) // Return the updated ApplicationFile
     }
+
+    pub async fn trigger_ssa(info: TriggerSSAInfo) -> Result<(), LDNError> {
+        let app_model =
+            Self::get_application_model(info.id.clone(), info.owner.clone(), info.repo.clone())
+                .await?;
+
+        let app_str = app_model.application.ok_or_else(|| {
+            LDNError::Load(format!(
+                "Application {} does not have an application field",
+                info.id
+            ))
+        })?;
+        let application_file = serde_json::from_str::<ApplicationFile>(&app_str).unwrap();
+
+        if application_file.lifecycle.state != AppState::Granted {
+            return Err(LDNError::Load(format!("Application state is {:?}. Expected Granted", application_file.lifecycle.state)));
+        }
+        let last_allocation = application_file.get_last_request_allowance().ok_or(LDNError::Load("Last allocation not found".into()))?;
+        if last_allocation.is_active {
+            return Err(LDNError::Load("Last active allocation ID is active".into()));
+        }
+
+        let requested_so_far = application_file.allocation.total_requested();
+        let total_requested = parse_size_to_bytes(&application_file.datacap.total_requested_amount).ok_or(
+            LDNError::Load("Can not parse total requested amount to bytes".into()),
+            )?;
+        
+        if requested_so_far >= total_requested {
```

Comment:
@Filip-L
```suggestion
        if requested_so_far + ssa_amount >= total_requested {
```
Where `ssa_amount` is info.amount converted to bytes.